homeassistant/components/light/tradfri.py

```diff
@@ -57,9 +74,26 @@ def brightness(self):
         return self._light_data.dimmer
 
     @property
+    def color_temp(self):
+        """Return the CT color value in mireds."""
+        if not self.supported_features & SUPPORT_COLOR_TEMP or \
+                not self._ok_temps:
+            return
+        kelvin = next((
+            kelvin for kelvin, hex_color in self._ok_temps.items()
+            if hex_color == self._light_data.hex_color), None)
+        if kelvin is None:
+            _LOGGER.error(
+                'unexpected color temperature found %s',
+                self._light_data.hex_color)
+            return
+        return color_util.color_temperature_kelvin_to_mired(kelvin)
+
+    @property
     def rgb_color(self):
         """RGB color of the light."""
-        return self._rgb_color
+        if self.supported_features & SUPPORT_RGB_COLOR:
```

Comment:
Shouldn't we always return this? The frontend will color the icon. We should not show the color picker if we do not support RGB.